test/test_torch.py

```diff
@@ -16614,10 +16615,12 @@ def inner(self, device, dtype):
     ('erfc', '', _small_3d, lambda t, d: [], 1e-3, 1e-5, 1e-5, _float_types),
     ('exp', '', _small_3d, lambda t, d: [], 1e-2, 1e-5, 1e-5, _float_types),
     ('expm1', '', _small_3d, lambda t, d: [], 1e-2, 1e-5, 1e-5, _float_types),
-    ('reciprocal', '', _small_3d, lambda t, d: [], 1e-1, 1e-5, 1e-5, _float_types),
-    ('floor', '', _small_3d, lambda t, d: [], 1e-5, 1e-5, 1e-5, _float_types),
-    ('frac', '', _small_3d, lambda t, d: [], 1e-5, 1e-5, 1e-5, _float_types),
-    ('neg', '', _small_3d, lambda t, d: [], 1e-5, 1e-5, 1e-5, _float_types2),
+    ('reciprocal', '', _small_3d, lambda t, d: [], 1e-1, 1e-2, 1e-5, _float_types),
+    ('reciprocal', 'no_zero', lambda t, d: _small_3d(t, d, has_zeros=False),
```

Comment:
Why do you need a special test for no_zero? `nan` results are correctly handled in comparisons.

Reply:
if we have zero number in input tensor, the output will have **inf** value for float32 and bfloat16, and then get a **nan** value(inf -inf), which report a error **nan is not less than 1e-2**.

Reply:
You can change the line `self.assertEqual(cpu_result, device_result, prec=precision, exact_dtype=False)` to 
```
self.assertEqual(cpu_result, device_result, prec=precision, exact_dtype=False, allow_inf=True)
```
in the test itself to handle this.